flict/flictlib/project/reader.py

```diff
@@ -200,9 +229,9 @@ def _flatten_package_tree(self, packages):
 
         for _package in packages.values():
             package_dict.update(
-                {f"{_package['name']}--{_package['version']}": _package},
+                {f"{_package['name']}--{_package.get('version', '')}": _package},
```

Comment:
As `version` is mandatory (IMHO) we should keep the old implementation

Reply:
https://spdx.github.io/spdx-spec/package-information/#73-package-version-field
* 7.3 Package version field, Required | No

Reply:
But (woops) the above is the internal format and version should be present (could be '').
Using `_package['version']`